### DISCUSSION TITLE
Integrate inverter from Solavita, model SP R8KH3 into Foxess Modbus

### DISCUSSION THREAD

**Discussion by Mike332332** [nathanmarlor/foxess_modbus#539] [Support] [answered]
Hello everyone,
I have a Solavita inverter, model SP R8KH3, in operation. It is identical in construction to Fox ESS H3. The Fox ESSCoud integration in Homeassistant already works for me. But I would like to use the FoxESS Modbus version. I connected via USB to RS485 adapter.
Would it be possible to integrate the Solavita SP R8KH3 model into FoxESS Modbus. Can someone help me or give me instructions on where and how I can register the inverter?
I get the following error message:
Error connecting to your adapter or inverter. Ensure the adapter is properly configured and is correctly wired to your inverter (see the setup link above), then try again. Details: Inverter model not supported: 'SP R8KH3'
I would be very grateful for a solution
Kind regards, Mike

**Comment by canton7** [comment, marked as answer]
I've had a shot at implementing this. Give the pre-release v1.10.0b10 a try, and let me know if anything doesn't work. Thanks!

**Reply by Mike332332** [reply]
Hello,
Thanks for the efforts, I have overwritten the 3 changed files, however I still get the same error. How can I install version v1.10.0b10 in homeassistant if I have version v1.10.0b9?
Sorry, I'm still a beginner :-)

**Reply by canton7** [reply]
HACS -> Integrations -> FoxESS - Modbus -> Hamburger menu in the top right -> Redownload, and turn on "Show beta versions".

Then follow the prompts to restart HA.

**Reply by Mike332332** [reply]
Then I get the following message:
"The version selection will be removed in a future release. If you need to install a specific version, you can do so through the Update Entities service call."

**Reply by canton7** [reply]
Screenshot?

**Reply by canton7** [reply]
How fun, I don't see that.

Anyway, you can still use the version selector to update to v1.10.0b10?

Alternatively, HACS has noticed that an update is available (I can see the blue arrow in the background), so you should just be able to click the big "Update" button.

**Reply by canton7** [reply]
(The message I was replying to has been deleted for some reason)

**Comment by Mike332332** [comment]
![Screenshot 2024-02-22 164757](https://github.com/nathanmarlor/foxess_modbus/assets/150423601/d143bebd-4ce9-48d6-b00a-43e0a0091d42)

**Reply by canton7** [reply]
You can still use the version selector to update to v1.10.0b10?

Alternatively, HACS has noticed that an update is available (I can see the blue arrow in the background), so you should just be able to click the big "Update" button.

**Comment by Mike332332** [comment]
![Screenshot 2024-02-22 165658](https://github.com/nathanmarlor/foxess_modbus/assets/150423601/9b0e4832-3943-43f4-bc13-c15739ac92cb)

**Reply by canton7** [reply]
Please just use HACS to update to v1.10.0b10.

**Reply by Mike332332** [reply]
I now have the correct version, but it still doesn't work

**Reply by canton7** [reply]
Did you restart HA properly (not just a "Quick Reload"), as prompted by HACS?

**Reply by Mike332332** [reply]
yes

**Comment by Mike332332** [comment]
![Screenshot 2024-02-22 170117](https://github.com/nathanmarlor/foxess_modbus/assets/150423601/814053c5-75fd-472c-a43b-922500c16fec)

**Reply by canton7** [reply]
Aha, I've spotted my error.

Try changing the following code in `foxess_modbus/custom_components/foxess_modbus/inverter_profiles.py`:

```py
        InverterModelProfile(
            SOLAVITA_SP,
            r"^R(\d+)KH3",
            capacity_map={
                "8": 10400,
                "10": 13000,
                "12": 15600,
            },
        ).add_connection_type(
            AUX,
            RegisterType.HOLDING,
            special_registers=H3_REGISTERS,
        ),
```

to
```py
        InverterModelProfile(
            SOLAVITA_SP,
            r"^SP R(\d+)KH3",
            capacity_map={
                "8": 10400,
                "10": 13000,
                "12": 15600,
            },
        ).add_connection_type(
            AUX,
            RegisterType.HOLDING,
            special_registers=H3_REGISTERS,
        ),
```

**Reply by Mike332332** [reply]
Thank you very much, I think it works, I am very happy and thank you for your help.

**Reply by canton7** [reply]
Great, thanks for checking. I'll include that fix in the next release.

Can you check that the Force Charge / Force Discharge work modes function?

**Comment by Mike332332** [comment]
![Screenshot 2024-02-22 180039](https://github.com/nathanmarlor/foxess_modbus/assets/150423601/1ce1d415-00cf-4107-88b8-39c254e2458d)

**Reply by Mike332332** [reply]
I think so, yes
